JAXB casting error?

Trying to use 2.0.2 unshaded and getting the following error.  In my Tomcat home/lib, I have jaxb-api.jar file, but seems the Docker client is trying to use jakarta.xml.bind-api-2.3.2.jar??
```
26-Nov-2019 09:47:32.537 SEVERE [RMI TCP Connection(3)-127.0.0.1] com.sun.jersey.server.impl.wadl.WadlApplicationContextImpl.<init> Provider class com.sun.xml.bind.v2.ContextFactory could not be instantiated: javax.xml.bind.JAXBException: ClassCastException: attempting to cast jar:file:/C:/software/tomcat9026/lib/jaxb-api.jar!/javax/xml/bind/JAXBContext.class to jar:file:/C:/software/tomcat9026/webapps/arcgis%23rest/WEB-INF/lib/jakarta.xml.bind-api-2.3.2.jar!/javax/xml/bind/JAXBContext.class.  Please make sure that you are specifying the proper ClassLoader.    	
	javax.xml.bind.JAXBException: Provider class com.sun.xml.bind.v2.ContextFactory could not be instantiated: javax.xml.bind.JAXBException: ClassCastException: attempting to cast jar:file:/C:/software/tomcat9026/lib/jaxb-api.jar!/javax/xml/bind/JAXBContext.class to jar:file:/C:/software/tomcat9026/webapps/arcgis%23rest/WEB-INF/lib/jakarta.xml.bind-api-2.3.2.jar!/javax/xml/bind/JAXBContext.class.  Please make sure that you are specifying the proper ClassLoader.    	
 - with linked exception:
[javax.xml.bind.JAXBException: ClassCastException: attempting to cast jar:file:/C:/software/tomcat9026/lib/jaxb-api.jar!/javax/xml/bind/JAXBContext.class to jar:file:/C:/software/tomcat9026/webapps/arcgis%23rest/WEB-INF/lib/jakarta.xml.bind-api-2.3.2.jar!/javax/xml/bind/JAXBContext.class.  Please make sure that you are specifying the proper ClassLoader.    ]
		at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:202)
		at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:129)
		at javax.xml.bind.ContextFinder.find(ContextFinder.java:307)
		at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:478)
		at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:435)
		at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:336)
		at com.sun.jersey.server.impl.wadl.WadlApplicationContextImpl.<init>(WadlApplicationContextImpl.java:107)
		at com.sun.jersey.server.impl.wadl.WadlFactory.init(WadlFactory.java:100)
		at com.sun.jersey.server.impl.application.RootResourceUriRules.initWadl(RootResourceUriRules.java:169)
		at com.sun.jersey.server.impl.application.RootResourceUriRules.<init>(RootResourceUriRules.java:106)
		at com.sun.jersey.server.impl.application.WebApplicationImpl._initiate(WebApplicationImpl.java:1359)
		at com.sun.jersey.server.impl.application.WebApplicationImpl.access$700(WebApplicationImpl.java:180)
		at com.sun.jersey.server.impl.application.WebApplicationImpl$13.f(WebApplicationImpl.java:799)
		at com.sun.jersey.server.impl.application.WebApplicationImpl$13.f(WebApplicationImpl.java:795)
		at com.sun.jersey.spi.inject.Errors.processWithErrors(Errors.java:193)
		at com.sun.jersey.server.impl.application.WebApplicationImpl.initiate(WebApplicationImpl.java:795)
		at com.sun.jersey.server.impl.application.WebApplicationImpl.initiate(WebApplicationImpl.java:790)
		at com.sun.jersey.spi.container.servlet.ServletContainer.initiate(ServletContainer.java:509)
		at com.sun.jersey.spi.container.servlet.ServletContainer$InternalWebComponent.initiate(ServletContainer.java:339)
		at com.sun.jersey.spi.container.servlet.WebComponent.load(WebComponent.java:605)
		at com.sun.jersey.spi.container.servlet.WebComponent.init(WebComponent.java:207)
		at com.sun.jersey.spi.container.servlet.ServletContainer.init(ServletContainer.java:394)
		at com.sun.jersey.spi.container.servlet.ServletContainer.init(ServletContainer.java:744)
		at org.apache.catalina.core.ApplicationFilterConfig.initFilter(ApplicationFilterConfig.java:270)
		at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:251)
		at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:102)
		at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4533)
		at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5172)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:717)
		at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:690)
		at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:705)
		at org.apache.catalina.startup.HostConfig.manageApp(HostConfig.java:1728)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at org.apache.tomcat.util.modeler.BaseModelMBean.invoke(BaseModelMBean.java:289)
		at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:809)
		at java.management/com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801)
		at org.apache.catalina.mbeans.MBeanFactory.createStandardContext(MBeanFactory.java:457)
		at org.apache.catalina.mbeans.MBeanFactory.createStandardContext(MBeanFactory.java:406)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at org.apache.tomcat.util.modeler.BaseModelMBean.invoke(BaseModelMBean.java:289)
		at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:809)
		at java.management/com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801)
		at java.management/com.sun.jmx.remote.security.MBeanServerAccessController.invoke(MBeanServerAccessController.java:468)
		at java.management.rmi/javax.management.remote.rmi.RMIConnectionImpl.doOperation(RMIConnectionImpl.java:1466)
		at java.management.rmi/javax.management.remote.rmi.RMIConnectionImpl$PrivilegedOperation.run(RMIConnectionImpl.java:1307)
		at java.base/java.security.AccessController.doPrivileged(Native Method)
		at java.management.rmi/javax.management.remote.rmi.RMIConnectionImpl.doPrivilegedOperation(RMIConnectionImpl.java:1406)
		at java.management.rmi/javax.management.remote.rmi.RMIConnectionImpl.invoke(RMIConnectionImpl.java:827)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:359)
		at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
		at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
		at java.base/java.security.AccessController.doPrivileged(Native Method)
		at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
		at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:562)
		at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:796)
		at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:677)
		at java.base/java.security.AccessController.doPrivileged(Native Method)
		at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:676)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
		at java.base/java.lang.Thread.run(Thread.java:834)
	Caused by: javax.xml.bind.JAXBException: ClassCastException: attempting to cast jar:file:/C:/software/tomcat9026/lib/jaxb-api.jar!/javax/xml/bind/JAXBContext.class to jar:file:/C:/software/tomcat9026/webapps/arcgis%23rest/WEB-INF/lib/jakarta.xml.bind-api-2.3.2.jar!/javax/xml/bind/JAXBContext.class.  Please make sure that you are specifying the proper ClassLoader.    
		at javax.xml.bind.ContextFinder.handleClassCastException(ContextFinder.java:111)
		at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:187)
		... 71 more
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,12 +2,18 @@
 
 ## 3.0.0 (unreleased)
 
+This release is mostly a Guava-exclusion release since several of its used features
+has been incorporated to Java 8+.
+
+Additionally, although not expected to cause any issues, the `com.fasterxml.jackson.module:jackson-module-jaxb-annotations`
+has been excluded and it stopped being pulled as a transitive dependency of `docker-client`.
+
 ### Breaking changes
 
 * `RegistryConfigs.Builder#addConfig` does not ignore null `RegistryAuth` values anymore
 * `HostConfig.Builder#appendBinds*` methods removed
 * `HostConfig.Builder#binds(final Bind... binds)` method does not ignore null binds anymore
-* `Immutable*` types removed from public getters and builders
+* `Immutable*` types removed from public fields
 * `EventStream` now implements `java.util.Iterator<Event>`
 * `EventStream#close` and `LogStream#close` can now throw `IOException`
 * `LogStream#attach(OutputStream, OutputStream, boolean)` method removed. Methods should

--- a/pom.xml
+++ b/pom.xml
@@ -51,6 +51,17 @@
     <version.plugin.spotbugs>3.1.12.2</version.plugin.spotbugs>
   </properties>
   
+  <dependencyManagement>
+  	<dependencies>
+  		<dependency>
+  			<groupId>com.fasterxml.jackson</groupId>
+        	<artifactId>jackson-bom</artifactId>
+        	<version>${version.jackson}</version>
+        	<scope>import</scope>
+        	<type>pom</type>	
+  		</dependency>
+  	</dependencies>
+  </dependencyManagement>
   <dependencies>
     <!--compile deps-->
     <dependency>
@@ -61,12 +72,16 @@
     <dependency>
       <groupId>com.fasterxml.jackson.jaxrs</groupId>
       <artifactId>jackson-jaxrs-json-provider</artifactId>
-      <version>${version.jackson}</version>
+      <exclusions>
+      	<exclusion>
+      		<groupId>com.fasterxml.jackson.module</groupId>
+      		<artifactId>jackson-module-jaxb-annotations</artifactId>
+      	</exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-databind</artifactId>
-      <version>${version.jackson}</version>
     </dependency>
     <dependency>
       <groupId>org.glassfish.jersey.core</groupId>

